eth/downloader/downloader.go

```diff
@@ -1317,27 +1322,31 @@ func (d *Downloader) fetchParts(deliveryCh chan dataPack, deliver func(dataPack)
 			// Send a download request to all idle peers, until throttled
 			progressed, throttled, running := false, false, inFlight()
 			idles, total := idle()
-
+			pendCount := pending()
 			for _, peer := range idles {
 				// Short circuit if throttling activated
-				if throttle() {
-					throttled = true
+				if throttled {
 					break
 				}
 				// Short circuit if there is no more available task.
-				if pending() == 0 {
+				if pendCount = pending(); pendCount == 0 {
 					break
 				}
 				// Reserve a chunk of fetches for a peer. A nil can mean either that
 				// no more headers are available, or that the peer is known not to
 				// have them.
-				request, progress, err := reserve(peer, capacity(peer))
+				request, progress, throttle, err := reserve(peer, capacity(peer))
 				if err != nil {
+					log.Info("Error reserving fetch", "err", err)
```

Comment:
Should this rather be a Warn or even an Error log? If this should never happen, I think Error is warranted to make it pop out.

Reply:
Hm, looking at it now, it doesn't look like we actually return any error there. So either it it should return an error, or we should remove the returnvalue

Reply:
The only thing that can happen is "index allocatio went beyond...", which is logged inside the queue. I'm not sure there's any point in bubbling it up. If so, we could remove the error return there, it seems like. Should be one or the other.

Reply:
As far as I can tell, throwing that error out from queue doesn't really make sense. It would abort the current sync, but I'm not sure that's correct. The `reserve` should not need to give an error, it should just tell give us tasks to fetch, and failing that, tell us if it's throttled or if some other peers might get a task. 
So I'm considering removing `err` from the return signature.....

Reply:
done